### ISSUE TITLE
Boundary integrals on meshes with discontinuous node representations

### ISSUE THREAD

**Issue by mlstowell** [mfem/mfem#735]
I have a mesh which is periodic in one direction but not in another.  Consequently, the Nodes GridFunction within the mesh provides a discontinuous L2 representation of the coordinates.  Now, I need to apply an absorbing boundary condition on the non-periodic portion of the mesh.  This is setup as follows:
```c++
a1_ = new ParBilinearForm(HCurlFESpace_);
a1_->AddDomainIntegrator(new CurlCurlIntegrator(*muInvCoef_));
a1_->AddDomainIntegrator(new VectorFEMassIntegrator(*massCoef_));
a1_->AddBoundaryIntegrator(new VectorFEMassIntegrator(*abcCoef_), abc_marker_);
a1_->Assemble();
```
This fails within `BilinearForm::Assemble()` because a suitable boundary element transformation cannot be created.  The problem seems to originate within `Mesh::GetBdrElementTransformation` in two places; first `FiniteElementSpace::GetBdrElementVDofs()` returns an empty array of vdofs because L2 spaces have no dofs on element boundaries, and second `FiniteElementSpace::GetBE(i)` returns NULL because the `L2_FECollection` does not allocate elements for lower dimensional entities.

I'm not sure how to address this issue.  Would it be possible to use the `Tr_Elements` rather then the usual `L2_Elements` for these discontinuous representations?  If so, is there something equivalent to  `FiniteElementSpace::GetBdrElementVDofs()` which would extract the necessary data to create the point matrix needed by the element transformation?

Thanks,
Mark

**Comment by v-dobrev** [comment]
We can add support for meshes with DG nodes in the method `Mesh::GetBdrElementTransformation`. This should be similar to the way the same issue is handled in `Mesh::GetFaceTransformation`. One thing to be careful about is that the face and boundary elements (stored separately internally) may have different orientations - we want `Mesh::GetBdrElementTransformation` to return the transformation of the boundary element, not the face element (the two may differ by a rotation/vertex permutation).

**Comment by v-dobrev** [comment]
Hey Mark,

I made a branch that should fix the above issue: https://github.com/mfem/mfem/tree/periodic-mesh-bdr-transform-fix (PR #811).

Give it a try when you have time and let me know how it goes.

**Comment by mlstowell** [comment]
Thanks, Veselin!
I'll try to look at this right away.
